src/Serilog/Formatting/Json/JsonFormatter.cs

```diff
@@ -316,141 +218,7 @@ void WriteJsonProperty(string name, object? value, ref char? precedingDelimiter,
         output.Write('"');
         output.Write(name);
         output.Write("\":");
-        WriteLiteral(value, output);
+        _jsonValueFormatter.Format(new ScalarValue(value), output);
```

Comment:
Allocation of `ScalarValue` on each property, not good.

Reply:
I suggest to add `internal` to `protected virtual void FormatLiteralValue(object? value, TextWriter output)` and call it directly from here.

Reply:
Sorry. Not from here, but from other places
![изображение](https://github.com/serilog/serilog/assets/21261007/9ea52909-e489-4d81-a0fd-2e3f46c8a289)

Then you need one more `WriteJsonProperty` private method.

Reply:
Good suggestion, thanks; I've internalized the idea that `JsonFormatter` is the legacy approach, but since we're still shipping it I should pay more attention 😅 - will follow up 👍

Reply:
Fixed; managed to clobber quite a lot of (now) unnecessary code, too 👍